src/MasterStakingClient.ts

```diff
@@ -0,0 +1,149 @@
+// @ts-ignore
+import { masterStaking as backend } from '@xbacked-dao/xbacked-contracts';
+import { Account } from './Account';
+import {
+  AccountInterface,
+  MasterStakingAdminParameters,
+  MasterStakingColdState,
+  MasterStakingHotState,
+  MasterStakingGlobalView,
+  MasterStakingLocalView
+} from './interfaces';
+
+export class MasterStakingClient extends Account {
+  readonly id: number | undefined;
+
+  constructor(params: AccountInterface, contractId: number) {
+    super(params);
+    this.id = contractId;
+  }
+
+  private parseColdState = (rawColdState: any): MasterStakingColdState => {
+    return {
+      stakingASA: rawColdState.stakingASA.toNumber(),
+      stakingASADecimals: rawColdState.stakingASADecimals.toNumber(),
+      rewardASAs: rawColdState.rewardASAs.map((asa: any) => asa.toNumber()),
+      rewardASADecimals: rawColdState.rewardASADecimals.map((decimals: any) => decimals.toNumber()),
+    }
+  }
+
+  private parseHotState = (rawHotState: any): MasterStakingHotState => {
+    return {
+      adminAddress: this.reachStdLib.formatAddress(rawHotState.adminAddress),
+      deprecateTimeout: rawHotState.deprecateTimeout.toNumber(),
+      deprecateAt: rawHotState.deprecateAt.toNumber(),
+      rewardRatios: rawHotState.rewardRatios.map((ratios: any) => ratios.toNumber()),
+      rewardRate: rawHotState.deprecateAt.toNumber(),
```

Comment:
@thinc5 
i think this should be `rawHotState.rewardRate.toNumber()`

Reply:
Good pickup Nimi!